### PR TITLE
[CAS-1004] Themeing MessageListHeaderView

### PR DESCRIPTION
**Pull request by bychkovdmitry**
https://stream-io.atlassian.net/browse/CAS-1004

### 🎯 Goal

Make `MessageListHeaderView` customizable via style transformer (`TransformStyle.messageListHeaderStyleTransformer`) and theme attribute (`StreamUiTheme.streamUiMessageListHeaderStyle`)

### 🛠 Implementation details

Overall, there are 3 ways to customize the look and feel of `MessageListHeaderView`:
 1.  Via `streamUiMessageListHeaderStyle` theme attribute
 2. Via `MessageListHeaderView` layout attributes (overrides pt.1)
 3. Via `TransformStyle.messageListHeaderStyleTransformer` (overrides pt.1 and pt.2)

### 🎨 UI Changes & Testing

1. Try overriding online text color in the theme of our sample app:

```
<style name="StreamTheme" parent="@style/StreamUiTheme">
      <item name="streamUiMessageListHeaderStyle">@style/MessageListHeaderTheme</item>
</style>


<style name="MessageListHeaderTheme" parent="StreamUi.MessageListHeader" >
        <item name="streamUiMessageListHeaderDefaultLabelTextColor">@color/stream_ui_accent_green</item>
</style>
```

2. Try overriding online text color via layout attributes (this should override corresponding theme attribute):

```
<io.getstream.chat.android.ui.message.list.header.MessageListHeaderView
        android:id="@+id/messagesHeaderView"
        android:layout_width="0dp"
        android:layout_height="wrap_content"
        app:layout_constraintEnd_toEndOf="parent"
        app:layout_constraintStart_toStartOf="parent"
        app:layout_constraintTop_toTopOf="parent"
        app:streamUiMessageListHeaderDefaultLabelTextColor="@color/stream_ui_accent_red" />
```        

### ☑️ Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] PR targets the `develop` branch
- [x] Changelog is updated with client-facing changes
- [ ] New code is covered by unit tests
- [x] Comparison screenshots added for visual changes
- [ ] Affected documentation updated (CMS, cookbooks, tutorial)
- [x] Reviewers added

### 🎉 GIF

giphy (7)

### CODE REVIEW

File: stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/header/MessageListHeaderViewStyle.kt [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+    public val showBackButton: Boolean,
+    public val showBackButtonBadge: Boolean,
+    @ColorInt public val backButtonBadgeBackgroundColor: Int,
+    public val showSearchingForNetworkProgressBar: Boolean,
+    public val searchingForNetworkProgressBarTint: ColorStateList,
+) {
+
+    internal companion object {
+        operator fun invoke(context: Context, attrs: AttributeSet?): MessageListHeaderViewStyle {
+            context.obtainStyledAttributes(
```
Comment (bychkovdmitry) [inline stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/header/MessageListHeaderViewStyle.kt:50]: Obtain individual attrs listed in `MessageListHeaderView` from view layout attrs, or from `streamUiMessageListHeaderStyle` style in our theme

**Review by adasiewiczr — COMMENTED** [review]
Looks good to me but let's wait for @ogkuzmin review

File: stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/header/MessageListHeaderViewStyle.kt (hunk 1 above)
Comment (ogkuzmin) [reply stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/header/MessageListHeaderViewStyle.kt:50]: Wow, it looks like black magic. If everything works so smooth there is no problem between XML overriding for theme/style

**Review by ogkuzmin — APPROVED** [review]

**Review by adasiewiczr — APPROVED** [review]

File: stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/header/MessageListHeaderViewStyle.kt (hunk 1 above)
Comment (samiuelson) [reply stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/header/MessageListHeaderViewStyle.kt:50]: Yup, that's what I needed 🥇

**Pull request merged**